pkg/apis/apps/v1beta2/defaults_test.go

```diff
@@ -416,13 +406,6 @@ func TestSetDefaultReplicaSet(t *testing.T) {
 			t.Errorf("unexpected object: %v", rs2)
 			t.FailNow()
 		}
-		if test.expectSelector != reflect.DeepEqual(rs2.Spec.Selector.MatchLabels, rs2.Spec.Template.Labels) {
```

Comment:
@kow3ns I spent some time thinking whether to set `obj.Spec.Selector` to `nil` or `&metav1.LabelSelector{MatchLabels: map[string]string{}}`. In the end, i chose `nil` for simplicity. As a result, the test to compare `rs2.Spec.Selector.MatchLabels` and `rs2.Spec.Template.Labels` became unnecessary and was removed.